static_src/stores/user_store.js

```diff
@@ -249,19 +310,20 @@ class UserStore extends BaseStore {
     return this._currentViewedType;
   }
 
-  _hasRole(roleToCheck, userType) {
-    const user = this.currentUser;
-    if (!user) return false;
-    if (!user[userType]) return false;
-    return !!(user[userType].find((role) => role === roleToCheck));
+  get isLoadingCurrentUser() {
+    return this._loading.currentUser === true;
   }
 
-  currentUserHasSpaceRole(role) {
-    return this._hasRole(role, 'space_roles');
-  }
+  hasRole(userGuid, entityGuid, roleToCheck) {
+    let wrappedRoles = roleToCheck;
+    if (!roleToCheck.length) {
+      wrappedRoles = [roleToCheck];
+    }
 
-  currentUserHasOrgRole(role) {
-    return this._hasRole(role, 'organization_roles');
+    const key = entityGuid;
+    const user = this.get(userGuid);
+    const roles = user && user.roles && user.roles[key] || [];
+    return !!roles.find((role) => wrappedRoles.includes(role));
```

Comment:
We should add documentation for this method, I was thinking if you passed an array of roles, the check would only succeed if the user has ALL roles, but this is checking for ANY role which is a big difference.